### PR TITLE
uploaded algorithms for 12_6, hashtables

### PR DESCRIPTION
**Pull request by jerrodq2**

### CODE REVIEW

File: Jerrod/Algorithms/12_6.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    HashMap.prototype.add = function(key, value){
+      var hash = key.hashcode()
+      var index = mod(hash, this.capacity)
+      if(Array.isArray(this[index])){
+        this[index].push([key, value])
+      } else{ //if index is empty, this creates the inner array with the key value pair already inside
+        this[index].push([ [key,value] ])
+      }
```
Comment (codyw9524) [inline Jerrod/Algorithms/12_6.html:20]: Have you tested this?  There seem to be quite a few bugs.

**Comment by jerrodq2** [comment]
No, but I amended the code to use this.table[index] instead of this[index] if thats what you mean. My most recent version was in  my pull request for algorithms 12_7 I believe,  but I couldn't get the hashcode function from the book to work so I was unable to test them all out.

**Comment by codyw9524** [comment]
Here's the hashCode function:
```
String.prototype.hashCode = function(){
	var hash = 0;
	if(this.length == 0) return hash;
	for(var i = 0; i < this.length; i++){
		char = this.charCodeAt(i);
		hash = ((hash << 5) - hash) + char;
		hash &= hash;
	}
	return hash;
}
```

**Comment by codyw9524** [comment]
here's the helper functions:
```
String.prototype.hashCode = function(){
	var hash = 0;
	if(this.length == 0) return hash;
	for(var i = 0; i < this.length; i++){
		char = this.charCodeAt(i);
		hash = ((hash << 5) - hash) + char;
		hash &= hash;
	}
	return hash;
}

function mod(input, div){
	return (input % div + div) % div;
}
```

**Pull request merged**